iam.tf

```diff
@@ -1,12 +1,12 @@
 resource "aws_iam_instance_profile" "ecs_profile" {
-  name = "tf-created-AmazonECSContainerProfile-${var.name}"
-  role = "${aws_iam_role.ecs-role.name}"
-  path = "${var.iam_path}"
+  name_prefix = "${replace(format("%.64s", replace("tf-ECSProfile-${var.name}-", "_", "-")), "/\\s/", "-")}"
```

Comment:
this is a good start but i think insufficient; what we need to do is figure out how long of a prefix Terraform will generate, and then truncate the `name_prefix` so that `name_prefix + prefix <= 64`

Reply:
Hmm yeah, looks like `name_prefix` is 26 characters long, I'll update the PR to account for that. Here's the open TF issue where folks working on the aws provider are talking about it too. https://github.com/terraform-providers/terraform-provider-aws/issues/625